packages/datagrid/src/components/RichCellEditor/components/CellEditorDatalist.component.tsx

```diff
@@ -0,0 +1,57 @@
+import React, { useCallback, useState } from 'react';
+import debounce from 'lodash/debounce';
+
+import { Datalist } from '@talend/react-components';
+
+import { AgGridCellValue } from '../../../types';
+
+interface CellEditorDatalistPropTypes {
+	onFilter: (search: string) => Promise<AgGridCellValue[]>;
+	onChange: (event: React.ChangeEvent<HTMLInputElement>, value: string) => void;
+	value: string;
+}
+
+function CellDatalist(props: CellEditorDatalistPropTypes) {
+	const { value: initialValue, onFilter, onChange } = props;
+
+	const [value, setValue] = useState(initialValue);
+	const [isLoading, setIsLoading] = useState(false);
+	const [options, setOptions] = useState([] as AgGridCellValue[]);
+
+	const handleUserInput = (inputValue: string) => {
+		setIsLoading(true);
+
+		onFilter(inputValue)
+			.then(setOptions)
+			.finally(() => {
+				setIsLoading(false);
+			});
+	};
+	const debouncedHandleUserInput = useCallback(debounce(handleUserInput, 300), []);
+
+	return (
+		<Datalist
+			// Using autoFocus because <Datalist /> does not forward ref, so no manual focus possible
+			autoFocus // eslint-disable-line jsx-a11y/no-autofocus
```

Comment:
```suggestion
			autoFocus // eslint-disable-line jsx-a11y/no-autofocus
```
😬 That's hardcore. Any Datalist rendering would steal focus away and mess up natural tab navigation, right?

Reply:
"autoFocus" is not pretty, I agree but I had no other solution as-is (see line above).
This is done to keep the current behavior.